src/main/java/br/unb/cic/analysis/df/pessimistic/PessimisticTaintedAnalysis.java

```diff
@@ -0,0 +1,144 @@
+package br.unb.cic.analysis.df.pessimistic;
+
+import br.unb.cic.analysis.AbstractAnalysis;
+import br.unb.cic.analysis.AbstractMergeConflictDefinition;
+import br.unb.cic.analysis.model.Conflict;
+import br.unb.cic.analysis.model.Statement;
+import soot.Body;
+import soot.Unit;
+import soot.Value;
+import soot.ValueBox;
+import soot.jimple.InstanceInvokeExpr;
+import soot.toolkits.graph.ExceptionalUnitGraph;
+import soot.toolkits.scalar.ForwardFlowAnalysis;
+
+import java.util.HashSet;
+import java.util.Set;
+
+public class PessimisticTaintedAnalysis extends ForwardFlowAnalysis<Unit, PessimisticTaintedAnalysisAbstraction> implements AbstractAnalysis {
+
+    private Body methodBody;
+    private Set<Conflict> conflicts;
+    private AbstractMergeConflictDefinition definition;
+
+    public PessimisticTaintedAnalysis(Body methodBody, AbstractMergeConflictDefinition definition) {
+        super(new ExceptionalUnitGraph(methodBody));
+        this.methodBody = methodBody;
+        this.conflicts = new HashSet<>();
+        this.definition = definition;
+        this.definition.loadSinkStatements();
+        this.definition.loadSourceStatements();
+        doAnalysis();
+    }
+
+    @Override
+    protected PessimisticTaintedAnalysisAbstraction newInitialFlow() {
+        return new PessimisticTaintedAnalysisAbstraction();
+    }
+
+    @Override
+    protected void flowThrough(PessimisticTaintedAnalysisAbstraction in, Unit unit, PessimisticTaintedAnalysisAbstraction out) {
+        Statement statement = createStatement(unit);
+        in.copy(out);
+        detectConflicts(in, statement);
+        kill(out, statement);
+        gen(out, statement);
+    }
+
+    protected Statement createStatement(Unit d) {
+        Statement.Type type = Statement.Type.IN_BETWEEN;
+
+        boolean isSource = this.definition.isSourceStatement(d);
+        boolean isSink = this.definition.isSinkStatement(d);
+        if (isSource && isSink) {
+            type = Statement.Type.SOURCE_SINK;
+        } else if (isSource) {
+            type = Statement.Type.SOURCE;
+        } else if (isSink){
+            type = Statement.Type.SINK;
+        }
+
+        return Statement.builder()
+                .setClass(methodBody.getMethod().getDeclaringClass())
+                .setMethod(methodBody.getMethod())
+                .setType(type)
+                .setUnit(d)
+                .setSourceCodeLineNumber(d.getJavaSourceStartLineNumber()).build();
+    }
+
+    protected void detectConflicts(PessimisticTaintedAnalysisAbstraction in, Statement statement) {
+        if (statement.getType() == Statement.Type.SOURCE_SINK) {
+            conflicts.add(new Conflict(statement, statement));
+        }
+
+        if (statement.isSink()) {
+            for (ValueBox use : statement.getUnit().getUseBoxes()) {
+                Value value = use.getValue();
+                Statement markedStatement = in.getMarkedStatement(value);

```

Comment:
@rafaelmotaalves achei confusa a logica: a partir de um valor retorna um marked stmt? nao faz muito sentido pra mim

Reply:
A ideia desse método é retornar o statement que modificou o valor caso esse valor esteja marcado no in. Vou tentar pensar em um nome melhor para esse método.